mlflow/tracking/fluent.py

```diff
@@ -155,7 +155,23 @@ def start_run(run_id=None, experiment_id=None, run_name=None, nested=False):
 
 
 def end_run(status=RunStatus.to_string(RunStatus.FINISHED)):
-    """End an active MLflow run (if there is one)."""
+    """End an active MLflow run (if there is one).
+
+    It removes the active run from the global variable keeping track of
+    active runs, unset the ``MLFLOW_RUN_ID`` environment variable and
+    send a termination signal with the ``MLFlowClient``, updating the
+    status on the tracking URI.
+
+    Note that ``end_run()`` is automatically called at exit time (when
+    your script completes).
+
+    In distributed situations, your master node may start a new MLFlow
+    run, forward the MLFlow ``run_id`` to an executor (using the
+    environment variable ``MLFLOW_RUN_ID`` for example), and terminate,
+    even though the executor has not completed. To avoid unwanted early
+    termination of the MLFlow run, use ``atexit.unregister(end_run)``
+    on the master node.
```

Comment:
@guillaumegenthial Rather than modifying the `atexit` callback, the recommended approach for managing the state of distributed runs is to use the MLflow Client API (https://mlflow.org/docs/latest/python_api/mlflow.tracking.html#mlflow.tracking.MlflowClient). Can we modify this guidance accordingly?